### PR TITLE
Compatibility with JDK 11 and M1 macs

### PR DESCRIPTION
**Pull request by nineinchnick**
Integration tests run on JDK 17.

Regular tests run on JDK 11. Running on newer JDK would require upgrading Spring.

### CODE REVIEW

**Comment by nineinchnick** [comment]
@przemekak @sopel39 @radek-starburst PTAL

**Comment by radek-starburst** [comment]
LGTM

**Review by radek-starburst — APPROVED** [review]

File: docs/getting-started/tests/.mvn/jvm.config [hunk 1]
```diff
@@ -0,0 +1 @@
+--add-opens java.base/java.lang=ALL-UNNAMED
```
Comment (sopel39) [inline docs/getting-started/tests/.mvn/jvm.config:1]: add comment why it's added.

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java [hunk 2]
```diff
@@ -106,19 +106,18 @@ public BenchmarkExecutionResult execute(Benchmark benchmark, int benchmarkOrdina
 
     private BenchmarkExecutionResult executeBenchmark(Benchmark benchmark, Optional<ZonedDateTime> executionTimeLimit)
     {
-        BenchmarkExecutionResultBuilder resultBuilder = new BenchmarkExecutionResultBuilder(benchmark);
-        List<QueryExecutionResult> executions;
-        try {
-            executions = executeQueries(benchmark, benchmark.getPrewarmRuns(), true, executionTimeLimit);
```
Comment (sopel39) [inline benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:112]: Can't `executeQueries` throw an exception? You moved it outside of `try` clause.

This looks like unrelated change. Undo or separate commit?

File: benchto-driver/src/main/java/io/trino/benchto/driver/listeners/BenchmarkServiceExecutionListener.java [hunk 3]
```diff
@@ -131,11 +131,15 @@ public Future<?> benchmarkFinished(BenchmarkExecutionResult benchmarkExecutionRe
     {
         return CompletableFuture.supplyAsync(() -> getMeasurements(benchmarkExecutionResult), taskExecutor::execute)
                 .thenCompose(future -> future)
-                .thenApply(measurements -> new FinishRequestBuilder()
-                        .withStatus(benchmarkExecutionResult.isSuccessful() ? ENDED : FAILED)
-                        .withEndTime(benchmarkExecutionResult.getUtcEnd().toInstant())
```
Comment (sopel39) [inline benchto-driver/src/main/java/io/trino/benchto/driver/listeners/BenchmarkServiceExecutionListener.java:136]: did it throw?

**Review by sopel39 — APPROVED** [review]
lgtm % comment

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java (hunk 2 above)
Comment (nineinchnick) [reply benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:112]: I moved it out, because there are listeners that rely on the result being started, which only happens after pre-warm queries are finished. These exceptions cover the original one, which makes it very hard to debug benchmark configs. This is easily reproducible when you mess up the variables and try to run an invalid query. Moving pre-warm queries outside of the try block makes the whole driver fail fast with an actionable exception. I think in this case we don't want to continue the benchmark, if the whole benchmark is invalid.

File: benchto-driver/src/main/java/io/trino/benchto/driver/listeners/BenchmarkServiceExecutionListener.java (hunk 3 above)
Comment (nineinchnick) [reply benchto-driver/src/main/java/io/trino/benchto/driver/listeners/BenchmarkServiceExecutionListener.java:136]: Yes

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java (hunk 2 above)
Comment (sopel39) [reply benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java:112]: > Moving pre-warm queries outside of the try block makes the whole driver fail fast with an actionable exception. I think in this case we don't want to continue the benchmark, if the whole benchmark is invalid.

Pre-warm failure should only fail that particular query being benchmarked and not the whole benchto run.

File: docs/getting-started/tests/.mvn/jvm.config (hunk 1 above)
Comment (nineinchnick) [reply docs/getting-started/tests/.mvn/jvm.config:1]: I don't think I can add a comment to this file. Isn't the commit message enough? If not, I'll remove it.

File: benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java (hunk 2 above)
Comment (nineinchnick) [reply benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java]: I reverted this change, we can figure this out later.

**Comment by nineinchnick** [comment]
@sopel39 PTAL, should be read to go now.

File: benchto-driver/src/main/java/io/trino/benchto/driver/listeners/BenchmarkServiceExecutionListener.java [hunk 4]
```diff
@@ -131,11 +131,15 @@ public Future<?> benchmarkFinished(BenchmarkExecutionResult benchmarkExecutionRe
...
-                        .addMeasurements(measurements)
-                        .build())
+                .thenApply(measurements -> {
+                    FinishRequestBuilder builder = new FinishRequestBuilder()
+                            .withStatus(benchmarkExecutionResult.isSuccessful() ? ENDED : FAILED)
+                            .addMeasurements(measurements);
+                    if (benchmarkExecutionResult.getUtcEnd() != null) {
+                        builder.withEndTime(benchmarkExecutionResult.getUtcEnd().toInstant());
```
Comment (sopel39) [inline benchto-driver/src/main/java/io/trino/benchto/driver/listeners/BenchmarkServiceExecutionListener.java:139]: nit: comment would be nice when it can be `null`

**Review by sopel39 — APPROVED** [review]
